template/src/core/network/Fetcher.ts

```diff
@@ -0,0 +1,122 @@
+import {isObject} from "../util/object/objectUtils";
+import {stringifySearchParams} from "../util/url/urlUtils";
+import FetcherError from "./FetcherError";
+import {fetchJSONMiddleware} from "./fetcherUtils";
+
+export type FetcherMiddleware<Argument = any, ResolveWith = any> = (
+  dataFromLastMiddleware: Argument
+) => Promise<ResolveWith>;
+
+interface FetcherConfig {
+  baseUrl: string;
+  initOptions?: RequestInit;
+  responseMiddlewares?: FetcherMiddleware[];
+  rejectMiddlewares?: FetcherMiddleware[];

```

Comment:
what are the use cases of reject middleware? 🤔

Reply:
For example in case of an error, you want to:
```
IF error code is 403:
  ignore
ELSE:
  throw sentry
```

Fetcher's middlewares are roughly the same thing with axios' interceptors: https://axios-http.com/docs/interceptors